File: myblog-server/src/main/java/com/site/blog/util/PageResult.java
package com.site.blog.util;

import lombok.Data;

import java.io.Serializable;
import java.util.List;

/**
 * 分页工具类
 *
 * @author 13
 * @qq交流群 796794009
 * @email 2449207463@qq.com
 * @link http://13blog.site
 */
@Data
public class PageResult implements Serializable {

    //总记录数
    private long totalCount;
    //每页记录数
    private int pageSize;
    //总页数
    private int totalPage;
    //当前页数
    private int currPage;
    //列表数据
    private List<?> list;

    /**
     * 分页
     *
     * @param list       列表数据
     * @param totalCount 总记录数
     * @param pageSize   每页记录数
     * @param currPage   当前页数
     */
    public PageResult(List<?> list, long totalCount, int pageSize, int currPage) {
        this.list = list;
        this.totalCount = totalCount;
        this.pageSize = pageSize;
        this.currPage = currPage;
        this.totalPage = (int) Math.ceil((double) totalCount / pageSize);
    }


}

File: myblog-jpa/src/main/kotlin/com/site/blog/util/PageResult.kt
package com.site.blog.util

import lombok.Data
import java.io.Serializable

@Data
class PageResult(//列表数据
    private val list: List<*>, //总记录数
    private val totalCount: Long, //每页记录数
    private val pageSize: Int, //当前页数
    private val currPage: Int
) : Serializable {
    //总页数
    private val totalPage: Int

    /**
     * 分页
     *
     * @param list       列表数据
     * @param totalCount 总记录数
     * @param pageSize   每页记录数
     * @param currPage   当前页数
     */
    init {
        totalPage = Math.ceil(totalCount.toDouble() / pageSize).toInt()
    }
}